File: python/MenuElem.py
class MenuElem:
    def __init__(self, name, subElems):
        self.name = name
        self.subElems = subElems
    def __cmp__(self, other):
        return self.name < other.name
    def __eq__(self, other):
        return self.name == other.name and self.subElems == other.subElems
    def __str__(self):
        return f"{self.name, self.subElems}"
    def __lt__(self, other):
        if self.name != other.name:
            return self.name < other.name
        return self.subElems < other.subElems

File: Lab16_3/MenuElem.cs
using System;
using System.Security.Authentication.ExtendedProtection.Configuration;
using System.Security.Cryptography;

namespace Lab16_3
{
    public class MenuElem:IComparable
    {
        public string name;
        public int subElems;

        public MenuElem(string name, int subElems)
        {
            this.name = name;
            this.subElems = subElems;
        }
        
        public int CompareTo(object obj)
        {
            var sho = (MenuElem) obj;
            if (this.name != sho.name)
                return this.name.CompareTo(sho.name);
            return this.subElems.CompareTo(sho.subElems); 
        }

        public override bool Equals(object obj)
        {
            var sho = (MenuElem) obj;
            return sho.name == this.name && sho.subElems == this.subElems;
        }

        public override string ToString()
        {
            return $"{this.name}, {this.subElems}";
        }
    }
}